Create liberland.md

flipchan opened the pull request:
liberland grant proposal

### Project Abstract

> Please provide a brief description of your project here summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [ ] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [ ] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [ ] The software delivered for this grant will be released under an open-source license specified in the application.
- [ ] The initial PR contains only one commit (squash and force-push if needed).
- [ ] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Grants-Program?pullRequest=831) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Grants-Program?pullRequest=831) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Grants-Program?pullRequest=831) it.</sub>

Noc2 reviewed (changes requested):
Thanks for the interest in our grants program. But the formatting of the application seems to be off, e.g. “ourownSubstratebasedblockchain.Theconsensusmechanismwillbea”. Also in general, it would be nice if you could use [our template ](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)and integrate the milestone as well as deliverables including the price per milestone etc. Furthermore it seems like you still need to sign the CLA.

takahser commented:
@flipchan thanks for the update, it looks better already. However, the milestone tables are still missing. Also, as mentioned by @Noc2 already, it would be good if you could stick to the structure of our [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) in general. Looking forward to the adjustments, I think this could become an exciting project!

Noc2 reviewed (changes requested):
Thanks for the update. But I agree with @takahser  it would be nice if you could stick to our template.

flipchan commented:
@Noc2 @takahser thanks for the feedback, I have updated the documents, let me know what I can add more or of it looks good

Noc2 reviewed (changes requested):
Thanks for the update. However, your application still doesn’t contain milestones with costs plus deliverables as well as the total costs. See https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules Also deliverables 0a-0d of the template are usually mandatory for all milestones

flipchan commented:
@Noc2 0a-0d added !

Noc2 reviewed (changes requested):
Thanks for integrating the mandatory deliverables. But it would help a lot if you could simply use the formatting of the template. Currently it’s for example unclear how much money you want to have per milestone or FTE, etc.? Also, a table helps us during the evaluation process and makes sure we don’t overlook anything.

flipchan commented:
@Noc2  added default formating for milestones, if there is anything else I can add, let me know

Noc2 reviewed (changes requested):
Thanks for the update. However, the application still is missing certain information, like for example the costs per milestone. Also your first milestone doesn’t seem to contain anything unique or new, unless I’m missing something. Keep in mind that our grants program focuses on funding open source software development that is beneficial for our ecosystem. We usually don’t support the deployment (of a substrate chain) at least not as the main goal of the delivery.

flipchan commented:
Hey @Noc2, I have updated the grant application based on your feedback!:) Let me know if it looks okey.

Looking forward to seeing this grant application be published and the Nation state of Liberland go on-chain!

takahser commented:
@flipchan your formatting is still off in large parts of the application because you formatted text as code, for example here:

![image](https://user-images.githubusercontent.com/5393704/158543612-cdbffbce-f82d-4600-b6cf-fd03665344a6.png)

Could you fix this?

Noc2 reviewed (changes requested):
Thanks again for the update. However, I still can not see the costs per milestone or the FTE. Additionally your milestones still contain a lot of deliveries that are simply done by cloning substrate and deploying it as a chain (see https://github.com/paritytech/substrate/tree/master/frame).  Again we don’t fund a deployment of a substrate chain, we only fund the development of unique pallets and technology that doesn’t exist already. So could you please remove these deliveries and instead focus on the unique things, which you develop that don’t already exist. Also in case you update an existing pallet, please describe what's the difference between your implementation and the existing implementation.

flipchan commented:
> @flipchan thanks for the update. Your formatting is still off in large parts of the application because you formatted text as code, for example here:
> 
> ![image](https://user-images.githubusercontent.com/5393704/158543612-cdbffbce-f82d-4600-b6cf-fd03665344a6.png)
> 
> Could you fix this? Removing [L291](https://github.com/w3f/Grants-Program/pull/831/files#diff-407622b7a9a07b77406d1bfbe02d81142678a43931b171c74a0ceb0509acba25R291) might be sufficient already (I didn't test it though).

Fixed!

takahser reviewed (changes requested):
@flipchan thx for the update. As [pointed out](https://github.com/w3f/Grants-Program/pull/831#pullrequestreview-911232663) by @Noc2, you're still missing the costs per milestone or the FTE. Could you add these pls?

flipchan commented:
@takahser added the total funding request amount on line 10(https://github.com/flipchan/Grants-Program/blob/patch-1/applications/liberland.md?plain=1#L10) :)

flipchan commented:
https://github.com/w3f/Grants-Program/pull/831/files#diff-407622b7a9a07b77406d1bfbe02d81142678a43931b171c74a0ceb0509acba25R10

Noc2 reviewed (changes requested):
@flipchan Thanks for pointing this out. But we need you to use the official template and add the costs per milestone. Otherwise, this will lead to confusion and might cause problems with our official Terms and Conditions. For example, we usually pay you after each milestone delivery and it's currently not clear how much we would pay you after milestone. It would be really appreciated if you could use the official template as much as possible. 

Regarding your delivery, please only focus on the new/unique technical development, which you do as part of the grant. We don’t support the deployment of a testnet or simply deploying/reusing code without any additional development. For example, in your first milestone it looks like you simply want to reuse parity’s stacking, transaction module, etc. without any changes.

semuelle commented:
- The Google Doc link under Liberland Merit seems to be invalid. 
- As Noc2 pointed out, you are still missing the duration and cost of each milestone, as shown in our template:

```
- **Estimated duration:** _ hours/weeks/months
- **FTE:** _
- **Costs:** _ USD
- ```

flipchan commented:
@Noc2 @semuelle @takahser I have updated the format based on the template provided

flipchan commented:
> Thanks again for the update. Looks much better now. However, could you be more specific regarding the new features that you plan to implement as part of the specification. For example, how is “User Roles” going to work and how is it different from https://github.com/paritytech/substrate/tree/master/frame/identity What is so unique about the distribution mechanism of the LLM currency? And what kind of voting mechanisms are you going to use that are not already covered by the existing pallets? Furthermore your second milestone seems similar to https://github.com/bit-country Did you already take a look at their work and how is it different from yours? Also feel free to remove the deliverables that don’t develop anything unique, like “Nominated Proof of stake consensus“. I assume you will simply use grandpa and babe for this, correct?

Hey, thanks for the feedback, I have added the documentation for the requested information, let me know if I can add anything more

flipchan commented:
@Noc2 
> What is so unique about the distribution mechanism of the LLM currency? 
The uniqueness is that this currency will have a hard limit, The LLM pallet will mint 0.9% of the total supply per year and so automatically using a pallet that is added to the runtime that runs a checking function on each block, after one year has past(X amount of blocks) the pallet will mint more tokens, following the google spreadsheet that is linked :)

> And what kind of voting mechanisms are you going to use that are not already covered by the existing pallets?    
I have added more documentation about this. We allow different user to vote on different things, for example a citizen will vote on a law proposal and a minister will vote on the budget proposal for what company or entity that will receive funding for building roads and similar activities. Added in the documentation

> Furthermore your second milestone seems similar to https://github.com/bit-country Did you already take a look at their work and how is it different from yours?   
I have added a comparison into the grant application about this :)

> Also feel free to remove the deliverable that don’t develop anything unique, like “Nominated Proof of stake consensus“. I assume you will simply use grandpa and babe for this, correct?

Removed the npos milestone, that is correct! Using the standard grandpa and babe. And removed extra non-unique milestones

Fixed :)

If anyone else has any feedback, let me know :) @takahser @semuelle

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+| 0e. | Article | We will publish a detailed article, walking a user through all functionality and features.)
+| 1. | Substrate chain | We will customize a substrate based code base and add our custom pallets |  
+| 2. | LLM module | We will create substrate pallet to handle the distribution and inflation of the LLM currency |  
+| 3. | Public Veto Module | We want our users to utilize our chain to have election and vote using a pallet  |  
```
Did you consider reusing the [democracy pallet](https://crates.io/crates/pallet-democracy/3.0.0/dependencies) instead? If not, how do they differ? In general, it'd be great to have more information about what kind of logic this pallet will encapsulate.

flipchan replied:
@takahser touched on this in the "Voting" section above, we want to use the vote functionality from democracy pallet and then add custom logic to allow certain users to vote on certain things

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+| 1. | Substrate chain | We will customize a substrate based code base and add our custom pallets |  
+| 2. | LLM module | We will create substrate pallet to handle the distribution and inflation of the LLM currency |  
+| 3. | Public Veto Module | We want our users to utilize our chain to have election and vote using a pallet  |  
+| 4. | Liberland Identity Pallet | We will create a substrate pallet that will allow accounts to have different account roles | 
```
Also here, there might be quite some overlap with the [identity pallet](https://crates.io/crates/pallet-identity). Could you give us a bit more information on how your pallet would differ from it?

flipchan replied:
Added a more in dept description and linked to the official documentation we have written for it

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+| 0d. | Docker | We will provide a Docker files that can be used to run our node and also test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish a detailed article, walking a user through all functionality and features.)
+| 1. | Land owner Pallet | We will create a pallet that will manage on-chain assets for land ownership |  
+| 2. | Metaverse integration Pallet | We will create a substrate pallet that interacts with our Metaverse|
```
Can you tell us a bit more about your metaverse? Especially in terms of technologies/networks/ecosystem(s) you're planning to use here.

flipchan replied:
added more information about it and added a youtube link people can view it in

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+We have currently had a very talented architect firm design and create our metaverse, that you can find more information about here:
+https://world.liberland.org/
+
+Our metaverse is currently being used for Virtual Reality Based conferences and social gatherings.
```
Your metaverse is already live?! I didn't find any information about it on your [webpage](https://world.liberland.org/
).

flipchan replied:
I added more documentation about this :)

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+
+### Long term vision
+
+The Liberland blockchain is a project for the long term. We aim to onboard our current user-base, which is currently over half a million users to use our chain for the majority of Liberland's e-government services.
```
```suggestion
The Liberland blockchain is a project for the long term. We aim to onboard our current user base, which is currently over half a million users to use our chain for the majority of Liberland's e-government services.
```

takahser reviewed (changes requested):
@flipchan thanks for the updates. Currently I'm not entirely clear on the functionality of your pallets. It would be nice if you could add additional details to all of them, so we can get a clearer picture. I also added a couple of more specific comments to some of them, so feel free to reply to these as well. 👍

flipchan commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+We name our currency and main on-chain token to Liberland Merit (LLM), signifying 
+the connection between money and real-life value. LLM will be the governance token
+of our blockchain and of Liberland itself. 
+The tokennomics of LLM will be a hard cap of 70 million being distributed over 
```
I reformulated the sentence

flipchan commented:
> @flipchan thanks for the updates. Currently I'm not entirely clear on the functionality of your pallets. It would be nice if you could add additional details to all of them, so we can get a clearer picture. I also added a couple of more specific comments to some of them, so feel free to reply to these as well. +1

All changes have been added :) Let me know if I can add some more changes

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,251 @@
+Based on a time based block production formula. This will allow the pallet to have time limits between the minting processes, allowing us to precalculate the circulating supply and the inflation curve. 
+
+Please see more details here:
+https://docs.google.com/spreadsheets/d/1uBuGNQMuyz9p4_JBbza7Wa8hWCYkGdGjoOI
```
```suggestion
https://docs.google.com/spreadsheets/d/1uBuGNQMuyz9p4_JBbza7Wa8hWCYkGdGjoOISF0LPKro/edit#gid=
```

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,251 @@
+
+Please see more details here:
+https://docs.google.com/spreadsheets/d/1uBuGNQMuyz9p4_JBbza7Wa8hWCYkGdGjoOI
+SF0LPKro/edit#gid=
```
```suggestion
```

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,250 @@
+
+[![IMAGE ALT TEXT](http://img.youtube.com/vi/G_1m8ZwqKMw/0.jpg)](http://www.youtube.com/watch?v=G_1m8ZwqKMw "Liberland MetaVerse")
+
+Our metaverse is not currently public, however, we are more then happy to provide a guided private tour of it, if requested.
```
I'd be interested in that.

flipchan replied:
We can schedule that if, if we can coordinate on times. Added an email to contact if needed :)

flipchan replied:
info on how to schedule it added

takahser replied:
I'll get back to you :) Thanks

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,241 @@
+https://docs.google.com/spreadsheets/d/1uBuGNQMuyz9p4_JBbza7Wa8hWCYkGdGjoOI
+SF0LPKro/edit#gid=
+
+We will create our own custom pallet, so the chain can automatically control the LLM's monetary model. Using pallet-assets, limited minting and distribution. 
```
@flipchan ^

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,250 @@
+Our metaverse is not currently public, however, we are more then happy to provide a guided private tour of it, if requested.
+
+
+##### Voting   
```
The Voting, _[Land Registry](https://github.com/w3f/Grants-Program/pull/831/files#diff-407622b7a9a07b77406d1bfbe02d81142678a43931b171c74a0ceb0509acba25R96) _and_ [Judiciary](https://github.com/w3f/Grants-Program/pull/831/files#diff-407622b7a9a07b77406d1bfbe02d81142678a43931b171c74a0ceb0509acba25R101)_ sections each describe a specific pallet. In the deliverables you choose different terms for the same concept, e.g. _Public Veto Module_. Please either consolidate these so that the same term is used for the same concept in order to make it easier to understand for everybody involved (see also [Ubiquitous Language](https://martinfowler.com/bliki/UbiquitousLanguage.html)). Or alternatively directly embed these 3 sections into the corresponding deliverables.

takahser reviewed (changes requested):
@flipchan thanks for the changes, we're making good progress here. 👍
Please have a look at my latest comments and also the previous ones you didn't yet reply.
Also, I corrected a broken link and directly pushed the commits, I hope that's fine.

flipchan commented on applications/liberland.md:
```diff
@@ -0,0 +1,250 @@
+Based on a time based block production formula. This will allow the pallet to have time limits between the minting processes, allowing us to precalculate the circulating supply and the inflation curve. 
+
+Please see more details here:
+https://docs.google.com/spreadsheets/d/1uBuGNQMuyz9p4_JBbza7Wa8hWCYkGdGjoOISF0LPKro/edit#gid=
```
That is an extra column representing amount of issued citizenship's per year, I have rewritten it so now it says clearly what the important columns are :)

flipchan commented:
updated the mentioned comments :) @takahser

semuelle commented:
@flipchan, what is you and your team's relationship with Liberland? Are you officially commissioned to implement it, or is this something you are planning to submit to Liberland's administration for consideration later?

flipchan commented:
> @flipchan, what is you and your team's relationship with Liberland? Are you officially commissioned to implement it, or is this something you are planning to submit to Liberland's administration for consideration later?

@semuelle we are officially part of Liberland's software development team, this application is coming straight from Liberland. We are currently building liberland's blockchain with substrate and therefor want to apply for funding and also get a better relationship with Parity and the web3 foundation, so that we can be part of substrate as an ecosystem. Our goal is to get this grant application through and hopefully we can be part of substrate's builders program. Here is a short talk I had about Liberland's journey into substrate at the liberland anniversary: https://www.youtube.com/watch?v=qhYTPtI5h1k 

Me personally, I'm very motivated by this projects, because it is the fore-runner of digital nation states running in trust less environments

semuelle reviewed (changes requested):
Thanks, happy to hear. 

With regard to the contents of the grant, I agree with @Noc2 and @takahser; it's not quite clear what the novelty of some of these things are. Keep in mind that we look for usefulness for the greater ecosystem, not only for the applicant. Therefore, your deliverable specifications aren't making a good case for themselves. For example, "We want our users to utilize our chain to have election and vote using a pallet" is not a specification. "[...] a substrate pallet that allows anyone that is holding LLM to incorporate a company" also doesn't convey much to explain its technical novelty. Lastly, I can't tell from [your description of the identity pallet](https://github.com/liberland/liberland_substrate/blob/main/docs/liberland/identity.md) how it is different from the existing `pallet-identity`. 

I think you may have to either be much more specific about the novelty (note that I am not talking about Liberland or the use of Substrate for real world governance, but the *technical* novelty) of each deliverable, or shrink it down to what is actually novel.

semuelle commented:
ping @flipchan.

takahser reviewed (changes requested):
@flipchan thanks for updating, looks good now. The only open issue for me is the metaverse integration, which I currently don't fully understand (see also @semuelle's comment). We'll reach out to you for a metaverse demo, where we hope to get a better understanding of it and how it's going to integrate with your metaverse pallet. Also, feel free to include more information regarding this metaverse integration into the grants application, even ahead of the meeting we're going to have. 👍

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+
+
+#### Time-Based Minting    
+The LLM pallet will allow adopters to delay the minting process using a regular DateTime format. LLM pre-calculates time based on block time, allowing us to postpone the execution of issuing tokens. This functionality does not currently exist in the substrate ecosystem. 
```
What do you mean by _adopters_? Collators?
Also, I'm not clear about the benefits of having this additional _DateTime_ interface - is there a reason to not simply specify the block time and rewards per block or era, by pre-calculating the size of the block rewards for each range of blocks, similar to BTC?

flipchan replied:
formated a bit better

takahser reviewed (changes requested):
I'd like to comment on various things. First of all, for the record:
- The Liberland team hosted a tour on their metaverse with us, however, due to technical reasons it didn't work at the time.
- Later, they were able to fix it and I took a look by myself.
- My first impression on their metaverse:
  - They allow for avatar customisation on a number of levels:
    <img width="1452" alt="image" src="https://user-images.githubusercontent.com/5393704/173700588-4ecfdf68-0a63-4633-81f6-e46d15035432.png">
  - The metaverse currently consists of 1 building in which the avatar is free to move:
    <img width="1454" alt="image" src="https://user-images.githubusercontent.com/5393704/173700917-927bc721-15d8-4e72-b83b-a633385fd106.png">
- The Liberland team plans the following integrations between Substrate and their metaverse:
  - they plan to be able to handle land registries on-chain which should be represented within the metaverse as well
  - they plan to support payments for any kind of trade
  - they plan to tokenize assets within the metaverse, making them available as NFTs
  - they plan to implement e-citizenship that would be represented by a non-tradable NFT

Furthermore, I took another iteration on the whole doc and highlighted a number of things, asked questions and proposed some changes. Please have a look at my new comments below @flipchan. Also, please have a look at the comments I made 21d ago which are still unaddressed.

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+
+
+##### Proof of Commitment
+We are implementing the identity pallet and adapting it to Liberland’s legal system: Citizenship requires a semi-permanent pooling of LLM to prove the applicant’s commitment to the State. When you move to a different country, this asks something of you, and there are requirements; we are mirroring this principle on-chain.                   
```
Yes, the government looks on-chain to see whether the Wallet of the given Citizen contains at least locked 5k LLM. This is judged as a proof of sufficient commitment to the State to warrant Citizenship. This is one of the steps in the Citizenship Application process, and another step is KYC. There is also the Citizenship interview, which is fully off-chain.

flipchan replied:
I have formulated this better

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+The powers of the Executive in Liberland are to govern directly on-chain within the limits of Congress's laws and off-chain as well in the metaverse and later the real-life state.
+    
+    
+More complex State actors might create specialized parachains which would govern their municipalities or even byzantine central bureaucracies. Given that States generate traffic as a matter of course, such systems are very interesting for Validators and Nominators and virtually non-existent in the current space.
```
We don't mean it as in the "byzantine consensus" or the "Byzantine General's Problem". Instead, we use the colloquial meaning. Bureaucracy that is "Byzantine" is complicated, complex, hard to understand. What we want to say: We aim to support various approaches to State administration, even ones vastly different and more complex than our own.
Hence: Byzantine ~= Complex. Sorry for the confusion.

flipchan replied:
changed this

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+
+#### Treasury   
+The Treasury for LLM, unlike that of Polkadot and like treasuries of extant States, operates without the burn mechanism. LLD follows the same rules as Polkadot, and so there is a burn mechanism to offset the “infinite” (really: algorithmic) supply. 
+As the LLM tokenomics suppose a deflationary token, holding LLM in the treasury will help finance running the State. The System distributes a significant portion of the treasury among the Members of Congress. Through this, the tokens get directly into the hands of the Voters (Citizens). Using Pallet-llm the treasury is the owner and manager of several on-chain assets, allowing us to expand the ba ...
```
1. Genesis: pre-mint the LLM
2. Place it in the Reserve Fund Wallet as locked
3. Have an algorithm release X amount per block into the TREASURY
4. Have the Senate hold the wallet as MultiSig
5. Have the entire Senate in consensus as requirement to do something else with the Reserve Fund than watch the algorithm do the controlled per-block release.

Treasury:
1. Have the Senate hold the deciding power to release funds from the Treasury (not consensus)
2. Have the Congress accept Spending Proposals instead of the Council ~Treasury Law
3. Have approx one Treasury Law per year which specifies which funds go to which wallet; wallets here belong to the branches of the government;
4. Upon the Spending Proposal being accepted, have it go to the Senate
5. If the Senate doesn't release the Funds, the payment doesn't go through (but the Spending Proposal funds get returned to the Wallet of the proposing account)
6. Have the

flipchan replied:
The treasury manage on-chain assets, this a customized implementation for liberland's chain with the LLM Pallet

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+
+
+#### Company Registration Pallet    
+The company registration pallet works similar to government company registrar authorities. This pallet will handle the registration of companies in a truly decentralized way. Allowing anyone to apply for registering on-chain companies
```
The Registrar of Companies will be an Agency of the Government. It will evaluate the contents of the Application. Later, we might make this decentralized (i.e. Validators will judge it based on a game theory principle, can message you more details on Matrix), but at first, we opt for a centralized solution, as we favour incremental approach and want to first build and deploy the basis.

flipchan replied:
updated it

flipchan commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+| 0e. | Article | We will publish a detailed article, walking a user through all functionality and features.)
+| 1. | LLM Pallet | LLM is a pallet customized to serve custom functionality for onchain Assets managed by the treasury |  
+| 2. | Congress Pallet | We are building a substrate pallet handling a tokenized direct democracy; we test it in limited conditions and then deploy it in the production environment |  
+| 3. | We will create a pallet that will be tailored to serve the voting and approval logic for governance for Nation States.   |  
```
fixed, good notice

takahser replied:
@flipchan it's not fixed, did you push all changes?

takahser replied:
PING @flipchan

flipchan replied:
hey, sorry for the delays, fixed now

flipchan commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+We build the LLM into the system using the underlying pallet assets combined with customized logic. The LLM Pallet will be able to support the following functionality:
+
+#### Treasury Minting and Predefined Max cap   
+Using a simple runtime parameter, we allow developers to simplify the printing amount of tokens, the amount of percentage to issue every X amount of time, and the maximum amount of tokens to be minted and sent to the Treasury, allowing the Treasury to manage on-chain assets. 
```
The treasury pallet id which also can act as an origin or an accountid is the owner of the minted funds and the LLM pallet lets the senate approve transfers from the treasury to a user account

flipchan replied:
I have made this clearer

flipchan commented:
@takahser @semuelle thanks for the comments, the document has now been updated

takahser commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+LLM will be minted over a timeframe of 48 years, allowing inflation to get fixed (being able to pre-calculate future inflation percentages) and controlled over time. Ninety-five per cent of all LLM get minted on behalf of the Treasury. Still, most of the Tokens typically don't stay there for long - they will be distributed into politics using the Members of Congress by a system described below. P ...
+
+#### Senate approved transfers   
+We create a special organ, the Senate, adaptable by other adopters as they see fit (see below). One of the main purposes of this organ is to preside over Treasury spending. It is not a political organ per-se; instead, it is composed of the most committed (invested) individuals in the State and the syste, providing a game-theory based incentive to make rational decisions.   
```
I'm pretty sure that's a typo @flipchan

michalptacnik replied:
Looks like it, yes.

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+
+#### A Tokenized Parliament    
+States are internationally expected to run some form of representative democracy. The core of the legislative power implemented on our chain is Congress. We suggest other adopters will keep and adapt it.  
+The Congress is a customized pallet, using delegated voting functionality using the underlying Democracy pallet. Where the original Democracy pallet effects direct democracy, we focus on a faithful representation of a Representative democracy. By default, States and State-like actors currently tend towards this type of democracy, but it is underrepresented in current blockchain projects. 
```
Philosophically, we believe that it is good to outsource the "positive" legislature to experts. The people are better at checking existing systems for trouble or just comparing them with what they individually want. They might be less good at actually writing coherent legislative systems. This is analogous to you bringing a paper to a conference; the people there are, individually and collectively, capable of creating something new from scratch without your help. But you bring your paper, the "kick-off" anyways, to stimulate the conversation and start the discussion. So - the representative democracy guarantees the quality, the coherence and the competence, and the people guarantee the legitimacy and provide the feedback loop.
At the end of the day, we have direct democracy: you can punish your representative for not doing what you want. You can join with others to be more effective. Then, you can submit to your representative your own legislative draft and give him the choice to either adopt and propose it, or lose the votes of your party/clique. This is not that different from how direct democracy would work.

Besides that, it is a "fixed requirement" to have a "government responsible to a democratically elected parliament." Would I say that there are no possible ways to satisfy this requirement with pure direct democracy? Probably not. But most states have parliaments, and this is what is expected. We are radical in many other ways (tokenized elections!), so we choose to fight battles we find important to win.

Makes sense?

takahser replied:
@flipchan

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+Representative democracy enacts laws and other regulations, both on-chain (as governance) and off-chain (for the state), and Congress is the heart of Liberland's political system. 
+
+#### Smart Elections   
+We implement an election system where a Citizen votes with their tokens by Liberland's design for a proof-of-stake-like meritocracy. The Citizens are analogous to Nominators. Instead of Validators, one chooses one or more Members of Congress to delegate their political power. 
```
We are building a system where Congress should get funding from the Treasury. This funding will be spread over the MoCs. As the MoCs are sitting in Congress, thanks to the voters, they will have the on-chain option to incentivise voting by offering returns to their voting for the given election period. 
Ever seen a system like that? Sure you did - this is how validators work on Polkadot, and that is, obviously, also our inspiration. So I used the metaphor of a proof-of-stake-like meritocracy. Voters are like nominators; MoCs are like validators. They take commission and spread profits amongst the voters. Just like validators would.
Jokingly, we call it legal bribing, but it is an incentive system that adds something our current democracy lacks - quantifying profit motive. Say that you have voters who only vote to get X and Y for themselves, the society can burn.
Conversely, you have voters who have a conception and want to make their ideas a reality. Usual people are between those two extremes. Our system provides a way to "equalize" along game-theory predictable (semi-predictable) lines the "profit-motivated" voters or a part of them.
The die-hard profit-minded voters of populism will be satisfied by going with the bigger parties (or MoCs), getting direct profits, quantifiable. They will thus follow a certain predictable curve, enabling the strongest MoCs to push through their policy. As we know from game theory, there is a correlation between voting by many people and the opinion quality. It is not a "given", for sure, and it is a VERY dangerous assumption in many systems, but it holds in general. Our democracies are not built on anything else in their cores. Majority voting is there primarily for legitimacy but also for this that we consider the majority opinion to be likelier the more correct one. See the American and English "majority" voting systems.

... makes sense?

takahser replied:
@flipchan

takahser replied:
Yes, thanks for elaborating! Appreciated

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+## Legislative Power 2: Direct democracy
+To fit the Liberland specs and enable more widespread adoption by traditional polities, we implement direct democracy. There, the system counts KYCed voters on the regular "one person-one vote" principle to introduce equality and equity between the rich and the poor (in Liberland, those who commit less and those who commit more).     
+
+Direct democracy is our basic check-and-balance: the people can second-guess their representatives in Congress. In Liberland, this is used to strike down unwanted regulations but can be easily adapted for positive democracy or even as the basis for elections. We complement this system with a front end.   
```
Kind of :) 
Voters are ALWAYS free to practice direct NEGATIVE democracy themselves. Your frustration, anger, knowledge, erudition and economic situation is more than sufficient to correlate the right incentive/skin in the game to decide what Law/Regulation/Policy is NOT good for you.
They can also play the "spot the crook" game by practising Dismissal; that is, public veto for elected officials, such as the judges. We trust that people will be able to define who in their mind has the over-critical amount of bad press, suspicious deals and personal affairs that they might want them no longer to represent Liberland.

If you want to engage in POSITIVE direct democracy, you join with others, form a party. Then you get your person voted in. Unlike anywhere else in the world, you have that person on an "algorithmic (on-chain) leash" - if they break your trust, they lose your confidence. It's over for them now (losing power to enact proposals) and it will sure be over come the next elections (in three months!).
Hence, our representative democracy is more direct than practically any direct democratic system. It is also really representative. As I write in the Grant Application, in today's states, you can't give any binding feedback mid term to a member of parliament who broke your trust. You are, in fact, choosing your masters, not your representatives. In liberland, they will represent you for real, and will get "slashed" by you if not.

I would just call what we do "democracy". Without adjectives. And that what other states do "something else. " :) Makes sense?

takahser replied:
Much clearer now, I was confused by the terms that apparently mean something else in the Liberland ecosystem than they do in existing countries. Thanks for clearifying!

michalptacnik commented on applications/liberland.md:
```diff
@@ -0,0 +1,345 @@
+We create a special organ, the Senate, adaptable by other adopters as they see fit (see below). One of the main purposes of this organ is to preside over Treasury spending. It is not a political organ per-se; instead, it is composed of the most committed (invested) individuals in the State and the syste, providing a game-theory based incentive to make rational decisions.   
+
+#### Incentive to HODL LLM   
+Due to the Proof of Commitment and the legislative system offering possible APR (see above and below respectively), token holders have an incentive to keep their LLM vested, and we expect this to raise and stabilize the price. 
```
Returns on investment.

takahser reviewed (changes requested):
Thanks for the update @flipchan @michalptacnik 👍
Please have a look at my not yet resolved comments and feel free to address them as well.

flipchan commented:
@takahser fixed the requested formating :)

flipchan commented:
poke @takahser

takahser reviewed (dismissed):
Thanks @flipchan for the updates.
I'm now happy with the scope of your grant application and grade of documentation on your pallets, so I'm willing to go ahead with it. I'm also going to share it with the team.

flipchan commented:
Looking forward to your response @Noc2 @semuelle @takahser

Noc2 reviewed (changes requested):
Thanks for chasing me and sorry for the late reply. The application and milestones look indeed much better now. I have a few additional comments:

- Could you remove all the parts from the application that specifically are about your tokenomics? For example: “Incentive to HODL LLM”,  “Liberland Dollar (LLD)” or “The hard cap of LLM will be 70 million.”. We are generally interested in supporting the development of open source pallets that potentially can be reused by others. The tokenomics might make some people believe that we supported these as part of application and we don’t want to create any confusion. 
- How does the Land owner pallet work? Is it similar to erc721/Nfs?

michalptacnik commented:
The Land Pallet will work similar to ERC NFTs, yes. We will register the Asset to become commercial on-chain as an NFT. It will have several properties too, and (ultimately), you will be able to attach one NFT to another in a simple chain. The point is to make it just complex enough to facilitate:
1) the sale and renting of virtual Assets like land, virtual houses, spaces, etc.
2) the sale and renting of real Assets which will be tokenized and will exist within our Assets-marketplace environment; and
3) the representation of collectables, the scarce or unique Assets in that world; on-chain. For collectors, but also for daily use. There will be many Liberland-specific examples, like having a semi-soulbound Citizenship NFT, which remembers who the previous owners had been.

We are devising a real-life Assets marketplace system in the virtual world. To tokenize something in the real world, to put a QR code to the physical thing; then you exchange the title in the virtual world and get an obligation to pay the money (the purchaser) and transfer the Asset (the seller). We are planning to create a part of our judicial system uniquely suited to adjudicate any disputes out of this, using a Kleros- or Julex-like system. But that is one of the further steps down the road.

flipchan commented:
@Noc2 application updated

Noc2 reviewed (approved):
Thanks for the update. I’m happy to go ahead with it.

semuelle reviewed (approved):
Thanks for taking the time to write incorporate so many updates, @flipchan & @michalptacnik. Happy to proceed. Just two side questions: 

- Why did you decide to remove the smart contract capabilities from the proposal? It seems to me that adding the contracts pallet to the chain spec would be little extra work. 
- How does one become a Liberland citizen? My understanding of a plain old nation state is that there is a central authority granting citizenship, which, in your case, would then be reflected on-chain. Would that be accurate?

michalptacnik commented:
<!--
/* Font Definitions */
@font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;}
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
/* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	text-decoration:underline;}
.MsoChpDefault
	{mso-style-type:export-only;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:70.85pt 70.85pt 70.85pt 70.85pt;}
div.WordSection1
	{page:WordSection1;}
/* List Definitions */
@list l0
	{mso-list-id:739866661;
	mso-list-template-ids:-1;}
@list l0:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Symbol;}
@list l0:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:"Courier New";
	mso-bidi-font-family:"Times New Roman";}
@list l0:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l0:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l0:level5
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l0:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l0:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l0:level8
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l0:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:324.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
ol
	{margin-bottom:0cm;}
ul
	{margin-bottom:0cm;}
-->Dear Sebastian, Thank you for your question! Smart Contracts: When writing the Grant Application, we stressed the importance of novel and Liberland-specific features, as instructed by your colleagues. We plan to introduce Smart Contracts to all parts of our system, as well as have bridges to other chains. We, however, recognize that this would hardly be a novel element as requested by your colleagues and we didn’t list it as one of its unique, core features. Despite all this, rest assured that we will have smart contracts in the same way other chains have them and in our unique ways as well (i.e. supporting the judiciary system). Granting/Removal of CitizenshipAs for Liberland citizenship, we currently have a centralized system, but we don't wish to keep it that way. Citizenship represents the recognition of one's bonds which tie them to a country by (ultimately) the other (earlier) citizens. Yes, current States do centralize this in their governments, but that is hardly the only thinkable option – and, we reason, it is decidedly not the best, most just and the most robust option. We haven't included this because we saw the grant application was already packed with features, and we didn't wish to make it leven onger. However, the idea is to create a decentralized system regarding the registrars. As part of this, or based on this, we will also allow certain collectives of citizens to confirm (validate) the KYC / personal profile of a prospective citizen in a fully decentralized manner. Removing citizenship will, ultimately, be in the hands of a special subsystem of the judiciary. While we favour the approval of one's peers for the *granting* of citizenship, we recognize the dangers of mob rule. Public international law also severely restricts citizenship removal, prohibiting the creation of Stateless persons (the taking away of one's last citizenship) outright and frowning upon the ostracisation of people possessing multiple passports. Hence, we will vest this responsibility in that organ best suited to see people as individuals and decide on their cases based on the rule of law. This will be both just in our opinion and internationally conform. Regarding how the judicial system is put together, we have judged this to be out of the scope of this current grant application, but we have spent a lot of time designing it. We want to make it public and share it with you as soon as we deem it sufficient to be relevant for the Substrate/Dotsama community. We namely envision this subsystem to be adaptable for parachains and other Substrate projects. I hope this answered your questions to your satisfaction. If you wish to ask anything more regarding this or other topics, please, let me know! Yours sincerelyMichal Ptacnik Verzonden vanuit Mail voor Windows Van: Sebastian MüllerVerzonden: dinsdag 5 juli 2022 13:02Aan: w3f/Grants-ProgramCC: michalptacnik; MentionOnderwerp: Re: [w3f/Grants-Program] Create liberland (PR ***@***.*** approved this pull request.Thanks for taking the time to write incorporate so many updates, @flipchan & @michalptacnik. Happy to proceed. Just two side questions:Why did you decide to remove the smart contract capabilities from the proposal? It seems to me that adding the contracts pallet to the chain spec would be little extra work.How does one become a Liberland citizen? My understanding of a plain old nation state is that there is a central authority granting citizenship, which, in your case, would then be reflected on-chain. Would that be accurate?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***> 

semuelle commented:
Hi @michalptacnik. Thanks for following up on my questions. I hope you are aware that your application was approved already?

flipchan commented:
@takahser @semuelle @noc2 the first milestone has been submitted:
https://github.com/w3f/Grant-Milestone-Delivery/pull/600
